### PR TITLE
Porting vmw-guestinfo to the magic world of 32 bits!

### PR DESCRIPTION
**Pull request by frapposelli**
It's pretty much that, nothing more.

### CODE REVIEW

File: vmcheck/vmcheck_linux_amd64.go [hunk 1]
```diff
@@ -48,12 +48,12 @@ func hypervisorPortCheck() (bool, error) {
 	}
 
 	p := &bdoor.BackdoorProto{}
-	p.CX.Low.SetWord(bdoor.CommandGetVersion)
 
+	p.CX.Low.SetWord(bdoor.CommandGetVersion)
 	out := p.InOut()
+	return (0xffffffff != out.AX.Low.Word()) && (0 != out.AX.Low.Word()), nil
 
 	// if there is no device, we get back all 1s
```
Comment (fdawg4l) [inline vmcheck/vmcheck_linux_amd64.go]: Small nit.  This comment is actually pretty useful and should probably go above the line at 54.

Comment (frapposelli) [inline vmcheck/vmcheck_linux_amd64.go]: 👍

File: examples/main.go [hunk 2]
```diff
@@ -34,13 +34,6 @@ func main() {
...
 
-	fmt.Println(config.SetString("foo", "bar"))
-	fmt.Println(config.String("foo", "foo"))
-
-	fmt.Println(config.SetInt("foo", 3))
-	fmt.Println(config.Int("foo", 0))
-
-	fmt.Println(config.SetBool("foo", false))
-	fmt.Println(config.Bool("foo", true))
+	fmt.Println(config.String("guestinfo.ovfEnv", ""))
```
Comment (dougm) [inline examples/main.go]: Probably want to undo this, and remove the binary examples/examples (add to .gitignore)

btw, maybe the example could be something like rpctool?  https://github.com/vmware/vic/blob/master/cmd/rpctool/main.go

**Comment by fdawg4l** [comment]
@dougm I wasn't sure if PCI addressing worked the same on other platforms and could only test on linux64.  Just FYI.

**Comment by frapposelli** [comment]
@dougm right now only vmcheck is constrained on `_linux`, the asm is constrained on the arch, and afaik there's no way to implement `hypervisorPortCheck()` without platform specific code, the rest is fine though.

I've replaced the example with `rpctool`, much better as a working example 😉

**Comment by sigma** [comment]
very nice. I'm a little bit concerned about the code duplication in 386/amd64 variants, but that can be addressed in a future cleanup PR

**Pull request merged**